### PR TITLE
[Enhancement] optimize planner lock contention 

### PR DESCRIPTION
**Pull request by murphyatwork**
## What I'm doing:
- Enable the PlannerOptimisticLock on follower node
- Enable the PlannerOptimisticLock for iceberg table

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 1]
```diff
@@ -169,20 +169,15 @@ public static ExecPlan planInsertStmt(PlannerMetaLocker plannerMetaLocker,
     private static boolean isLockFreeInsertStmt(InsertStmt insertStmt,
                                                 ConnectContext connectContext) {
         boolean isSelect = !(insertStmt.getQueryStatement().getQueryRelation() instanceof ValuesRelation);
-        boolean isLeader = GlobalStateMgr.getCurrentState().isLeader();
-        boolean isOnlyOlapTableQueries = AnalyzerUtils.isOnlyHasOlapTables(insertStmt);
-        return isOnlyOlapTableQueries && isSelect && isLeader &&
-                !connectContext.getSessionVariable().isCboUseDBLock();
+        boolean isOnlyOlapTableQueries = AnalyzerUtils.areTablesCopySafe(insertStmt);
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:172]: please change this variable name.

Comment (murphyatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java]: fixed

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 2]
```diff
@@ -125,7 +125,7 @@ public static ExecPlan plan(StatementBase stmt, ConnectContext session,
             if (stmt instanceof QueryStatement) {
                 QueryStatement queryStmt = (QueryStatement) stmt;
                 resultSinkType = queryStmt.hasOutFileClause() ? TResultSinkType.FILE : resultSinkType;
-                boolean isOnlyOlapTableQueries = AnalyzerUtils.isOnlyHasOlapTables(queryStmt);
+                boolean isOnlyOlapTableQueries = AnalyzerUtils.areTablesCopySafe(queryStmt);
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:128]: please modify here.

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 3]
```diff
@@ -169,20 +169,15 @@ public static ExecPlan planInsertStmt(PlannerMetaLocker plannerMetaLocker,
...
 
     private static boolean isLockFree(boolean isOnlyOlapTable, ConnectContext session) {
         // condition can use conflict detection to replace db lock
         // 1. all tables are olap table
-        // 2. node is master node
         // 3. cbo_use_lock_db = false
-        return isOnlyOlapTable
-                && GlobalStateMgr.getCurrentState().isLeader()
-                && !session.getSessionVariable().isCboUseDBLock();
+        return isOnlyOlapTable && !session.getSessionVariable().isCboUseDBLock();
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:180]: add here?

**Review by packy92 — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**